### PR TITLE
Remove Default FX_Version, as we let it to default to CLI in usage

### PR DESCRIPTION
**Pull request by ramarag**

### CODE REVIEW

**Comment by ramarag** [comment]
@eerhardt @gkhanna79 @nguerrera  PTAL

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets [hunk 1]
```diff
@@ -92,7 +92,7 @@ Copyright (c) .NET Foundation. All rights reserved.
                              _CGenJit=$(_JitPath);
                              _CGenInputAssembly=%(_AssembliestoCrossGen.Fullpath);
                              _CGenOutputAssembly=$([System.IO.Path]::Combine($(_RuntimeOptimizedDir),%(RecursiveDir)%(Filename)%(Extension)));
-                             _CGenPlatformAssembliesPath=$(_RuntimeRefDir)$(PathSeparator)$(_NetCoreRefDir);" />
+                             _CGenPlatformAssembliesPath=$(_NetCoreRefDir)$(PathSeparator)$(_RuntimeRefDir);" />
```
Comment (ramarag) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets]: making the netcoreapp as the first place to look for dependencies as we are no longer shipping the corefx libraries individually,

**Comment by piotrpMSFT** [comment]
How will this work for non-CLI builds? E.g. Standalone MSBuild, VS, VS for Mac?

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets (hunk 1 above)
Comment (gkhanna79) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets]: Is NetCoreRefDir pointer to folder of reference assemblies or implementation assemblies that you need for crossgen?

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets [hunk 2]
```diff
@@ -41,7 +41,6 @@ Copyright (c) .NET Foundation. All rights reserved.
       <PreserveCacheLayout Condition="'$(PreserveCacheLayout)' == ''">true</PreserveCacheLayout>
       <SkipOptimization Condition="'$(RuntimeIdentifier)' == ''">true</SkipOptimization>
       <_TFM Condition="'$(_TFM)' == ''">$(TargetFramework)</_TFM>
-      <FX_Version Condition="'$(FX_Version)' == ''">$(_ShortFrameworkVersion)</FX_Version>
```
Comment (gkhanna79) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets]: What happens if this SDK functionality is used in non-CLI scenarios (e.g. VS chooses to use it)? Who would set this?

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets [hunk 3]
```diff
@@ -152,7 +152,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       <_RuntimeRefDir>$([System.IO.Path]::Combine($(_CrossProjFileDir), "runtimeref"))</_RuntimeRefDir> <!-- flat app managed assemblies -->
       <_NetCoreRefDir>$([System.IO.Path]::Combine($(_CrossProjFileDir), "netcoreapp"))</_NetCoreRefDir> <!-- flat netcore app assemblies-->
       <_RuntimeOptimizedDir>$([System.IO.Path]::Combine($(_CrossProjFileDir), "runtimopt"))</_RuntimeOptimizedDir> <!-- optimized app managed assemblies in nuget cache layout -->
-      <_CrossProjFile>$([System.IO.Path]::Combine($(_CrossProjFileDir), "Optimize_Crossgen_$(RuntimeIdentifier)_$(_TFM)_$(FX_Version).csproj"))</_CrossProjFile>
+      <_CrossProjFile>$([System.IO.Path]::Combine($(_CrossProjFileDir), "Optimize_Crossgen_$(RuntimeIdentifier)_$(_TFM).csproj"))</_CrossProjFile>
```
Comment (gkhanna79) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets]: Why are we removing this? Doesn't `FX_Version` get set by the time we are here?

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets (hunk 1 above)
Comment (ramarag) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets]: i am using implementation assemblies here

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets (hunk 2 above)
Comment (ramarag) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets]: the user is supposed to set this, if not then SDK will deduce a default through RuntimeFrameworkVersion, which is what i am making it move to now.
previously was disabling the implicit reference and adding an explicit ref, but changing it as per @eerhardt 's   https://github.com/dotnet/sdk/pull/761#discussion_r98922463

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets (hunk 3 above)
Comment (ramarag) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets]: FX_Version is an optional parameter, previous deduction was not always safe

**Comment by ramarag** [comment]
merging this is as it is blocking the CLI integration. Will address any additional feedback later

**Pull request merged**